django/views/generic/dates.py

```diff
@@ -1,6 +1,7 @@
 import datetime
 
 from django.conf import settings
+from django.contrib.admin.utils import get_fields_from_path
```

Comment:
Some helper or something is needed because the current code is quite difficult to read and understand.